libyul/optimiser/Suite.cpp

```diff
@@ -139,7 +140,10 @@ void OptimiserSuite::run(
 			_optimizeStackAllocation,
 			stackCompressorMaxIterations
 		);
-	suite.runSequence("fDnTOc g", ast);
+
+	// Run the user-supplied clean up sequence
+	suite.runSequence(_optimisationCleanupSequence, ast);
+	suite.runSequence("g", ast);
```

Comment:
Why not let the user decide whether `g` is part of the cleanup sequence? (did not check the semantic of `g` yet)

Reply:
Because @ekpyron said so :)

Reply:
I think in general there should be no implicit settings, e.g. if a cleanup sequence was defined.. I'm also wondering about the part that is happening from line `123` on `suite.runSequence("hgfo", ast);`. What if someone really wants to run a sequence, where e.g. `hgfo` should explicitly not be done before? I could imagine that this could be important for testing purposes, but probably its also not needed at all.

Reply:
E.g. these are hardcoded parts of the optimizer sequence configurable that are not configurable.

Reply:
Maybe if an optimiser sequence is given by the user - no implicit settings should be used.

Reply:
``g`` is the ``FunctionGrouper`` - which just creates some canonical form of the AST without really changing much other than bringing the AST into a canonical form (``{ { I... } F... }`` with ``{ I... }`` is the main yul block and ``F...`` are all the functions). It has to be remain hardcoded here, since some optimizer steps break that form, but the StackCompressor and StackLimitEvader rely on it (and throw otherwise).

Reply:
Actually, can you add a distilled version of of what @ekpyron said as a comment? If this is not obvious to the reader, it should be explained.